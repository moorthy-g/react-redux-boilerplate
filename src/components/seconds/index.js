import Seconds from './components/seconds';
import reducer, { actions, STATE_KEY } from './state';

export default Seconds;
export { reducer, actions, STATE_KEY };
