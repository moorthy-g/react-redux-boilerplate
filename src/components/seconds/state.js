// State key
export const STATE_KEY = 'seconds';

// Action types
const UPDATE = 'SECONDS/UPDATE';

// Reducer
export default function (state = 0, action) {
  switch(action.type) {
    case UPDATE:
      return action.payload;
    default:
      return state;
  }
}

// Action creators
function updateSeconds() {
  return {
    type: UPDATE,
    payload: new Date().getSeconds()
  }
}

export const actions = {
  updateSeconds
}
