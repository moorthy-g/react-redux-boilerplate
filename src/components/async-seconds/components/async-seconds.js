import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { actions, STATE_KEY } from '../state';

class AsyncSeconds extends Component {
  componentDidMount() {
    this.props.updateSeconds();
    setInterval(this.props.updateSeconds, 2000);
  }
  render() {
    return ( <span>{this.props.seconds}</span> )
  }
}

function mapStateToProps(state) {
  return {
    seconds: state[STATE_KEY]
  }
}

function mapStateToDispatch(dispatch) {
  return bindActionCreators(actions, dispatch);
}

export default connect(mapStateToProps, mapStateToDispatch)(AsyncSeconds);
