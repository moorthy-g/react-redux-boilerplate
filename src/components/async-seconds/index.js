import store from 'store';
import AsyncSeconds from './components/async-seconds';
import { reducer, actions, STATE_KEY } from './state';

store.addReducer(STATE_KEY, reducer);

export default AsyncSeconds;
export { actions, STATE_KEY };
