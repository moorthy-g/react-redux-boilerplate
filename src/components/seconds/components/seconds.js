import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { actions, STATE_KEY } from '../state';

export class Seconds extends Component {
  componentDidMount() {
    this.props.updateSeconds();
    setInterval(this.props.updateSeconds, 5000);
  }
  render() {
    return ( <span>{this.props.seconds}</span> )
  }
}

function mapStateToProps(state) {
  return {
    seconds: state[STATE_KEY]
  }
}

function mapStateToDispatch(dispatch) {
  return bindActionCreators(actions, dispatch);
}

export default connect(mapStateToProps, mapStateToDispatch)(Seconds);
