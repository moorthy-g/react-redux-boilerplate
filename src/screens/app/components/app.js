import React from 'react';
import { connect } from 'react-redux';
import Loadable from 'react-loadable';
import Seconds from 'components/seconds';
import { actions, STATE_KEY } from '../state';
import '../style.less';

const AsyncSeconds = Loadable({
  loader: () => import('components/async-seconds'),
  loading: () => 'Loading...'
});

class App extends React.Component {
  render() {
    return (
      <div>
        <h2 styleName='title'>
          This is a react/redux webpack boilerplate <br />
          Seconds component <Seconds />
        </h2>
        <button onClick={this.props.showAsyncComponent}>Show Async Component</button>
        { this.props.asyncComponent &&
          <h2 styleName='title'>
            Async seconds component <AsyncSeconds />
          </h2>
        }
      </div>
    );
  }
}

function mapStateToProps(state) {
  return state[STATE_KEY];
}

export default connect(mapStateToProps, actions)(App);
