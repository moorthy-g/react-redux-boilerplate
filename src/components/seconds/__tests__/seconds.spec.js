import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Seconds } from '../components/seconds';

describe('<Seconds />', () => {
  it('should show seconds prop', () => {
    const wrapper = shallow(<Seconds seconds={10} updateSeconds={() => {}} />);
    expect(wrapper.find('span').text()).toEqual('10');
  });
  it('should match snapshot', () => {
    const output = renderer.create(<Seconds seconds={0} updateSeconds={() => {}} />);
    expect(output).toMatchSnapshot();
  });
});
