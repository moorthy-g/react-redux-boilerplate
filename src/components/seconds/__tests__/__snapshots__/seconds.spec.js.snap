// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Seconds /> should match snapshot 1`] = `
<span>
  0
</span>
`;
